orm/index.go

```diff
@@ -183,33 +216,61 @@ func (i Index) move(db weave.KVStore, prev Object, save Object) error {
 		return ErrModifiedPK()
 	}
 
-	// if the keys don't change, then
-	oldKey, err := i.index(prev)
+	oldKeys, err := i.index(prev)
 	if err != nil {
 		return err
 	}
-	newKey, err := i.index(save)
+	newKeys, err := i.index(save)
 	if err != nil {
 		return err
 	}
-	if bytes.Equal(oldKey, newKey) {
-		return nil
+
+	oldKeysSet := make(map[string]struct{})
+	for _, v := range oldKeys {
+		oldKeysSet[string(v)] = struct{}{}
 	}
 
-	// check unique constraint before removing
-	if i.unique && len(oldKey) != 0 {
-		k := i.IndexKey(newKey)
-		val := db.Get(k)
-		if val != nil {
-			return ErrUniqueConstraint(i.name)
+	newKeysSet := make(map[string]struct{})
+	for _, v := range newKeys {
+		newKeysSet[string(v)] = struct{}{}
+	}
+
+	// check unique constraints first
```

Comment:
maybe calculate `toAddSet` and `toRemoveSet` as diffs, then some of the below code is simplified